### PR TITLE
fix/get cities names and average salaries in country

### PR DESCRIPTION
**Pull request by Abdallahx3x**
exclude null salaries
make some changes to the tests

### CODE REVIEW

File: src/main/kotlin/interactor/GetCitiesNamesAndAverageSalariesInCountry.kt [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
+        val list=dataSource
             .getAllCitiesData()
-            .filter{it.country.lowercase()==country.lowercase()&&it.dataQuality}
+            .filter{it.country.lowercase()==country.lowercase()&&excludeNullSalariesAndLowQualityData(it)}
             .map{it.cityName to it.averageMonthlyNetSalaryAfterTax!!}
 
-        if (dataSource.isEmpty()) throw Exception("wrong name")
-
-        return dataSource
+        if (list.isEmpty()) throw Exception("wrong name")
```
Comment (Abdelrahman-Sedky) [inline src/main/kotlin/interactor/GetCitiesNamesAndAverageSalariesInCountry.kt:16]: Use scope functions to do the same logic of this line and return the result value directly without the need of the variable `list`.

File: src/test/kotlin/interactor/GetCitiesNamesAndAverageSalariesInCountryTest.kt [hunk 2]
```diff
@@ -67,7 +71,14 @@ class GetCitiesNamesAndAverageSalariesInCountryTest {
         //when get the cities name and average salary in country
         // then check the result
         assertThrows<Exception> {getResult.execute(country)}
-
+    }
+    @Test
+    fun should_ThrowException_when_EnterEmptyName() {
```
Comment (Abdelrahman-Sedky) [inline src/test/kotlin/interactor/GetCitiesNamesAndAverageSalariesInCountryTest.kt:76]: Use `` **backticks** in naming for better readability.
> EX: `should throw exception when given empty name`

File: src/test/kotlin/interactor/GetCitiesNamesAndAverageSalariesInCountryTest.kt [hunk 3]
```diff
@@ -67,7 +71,14 @@ class GetCitiesNamesAndAverageSalariesInCountryTest {
...
+        //given name of the country
+        val country=""
+        //when get the cities name and average salary in country
```
Comment (Abdelrahman-Sedky) [inline src/test/kotlin/interactor/GetCitiesNamesAndAverageSalariesInCountryTest.kt:79]: Better option
```suggestion
        //when get the cities name and average salary in country
        val ListOfCitiesAndSalaries = Executable { getResult.execute(country) }
        // then check the result
        assertThrows(Exception::class.java, ListOfCitiesAndSalaries)
```

**Review by Abdelrahman-Sedky — CHANGES_REQUESTED** [review]
Great job so far.
some little changes are required.

**Review by Abdelrahman-Sedky — APPROVED** [review]
LGTM

**Review by Abdelwahab-IO — APPROVED** [review]

**Pull request merged**